src/components/Dashboard/Dashboard.test.tsx

```diff
@@ -1,19 +1,176 @@
-import { render, screen } from '@testing-library/react'
 import React from 'react'
-import { BrowserRouter } from 'react-router-dom'
+import { render, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { BrowserRouter, MemoryRouter } from 'react-router-dom'
 import { Dashboard } from './Dashboard'
+import * as ImagesService from '../../services/ImagesService/ImagesService'
+import * as ImageRepositoryService from '../../services/ImageRepositoryService'
+import { SkipImageResponse } from '../../services/ImagesService/api/SkipImageApi'
+import { SnackbarProvider } from 'notistack'
 
 describe('Dashboard', () => {
-  it('should shown a canvas', () => {
-    render(
-      <BrowserRouter>
-        <Dashboard />
-      </BrowserRouter>,
-    )
+  describe('Canvas', () => {
+    it('should shown a canvas', () => {
+      render(
+        <BrowserRouter>
+          <SnackbarProvider>
+            <Dashboard />
+          </SnackbarProvider>
+        </BrowserRouter>,
+      )
 
-    const canvasTagMatcher = (_content: string, element: Element | null) => element?.tagName.toLowerCase() === 'canvas'
+      const canvasTagMatcher = (_content: string, element: Element | null) =>
+        element?.tagName.toLowerCase() === 'canvas'
 
-    const myCanvas = screen.getAllByText(canvasTagMatcher)
-    expect(myCanvas.length).toBeGreaterThan(0)
+      const myCanvas = screen.getAllByText(canvasTagMatcher)
+      expect(myCanvas.length).toBeGreaterThan(0)
+    })
+  })
+
+  describe('Actions', () => {
```

Comment:
I would suggest to add these tests under a different file, because it's getting a long a test file.

Reply:
I think the main point here is the size/responsibilities of the Dashboard component itself.
So I would say that for next changes it would be good to refactor it (and the tests respectively).

Thanks for highlighting it.